### DISCUSSION TITLE
Environment

### DISCUSSION THREAD

**Discussion by JosephEDale** [smartcontractkit/full-blockchain-solidity-course-py#1756] [Q&A]
On remix I am struggling to work out why I cant switch from "remix VM (London)" to Injected Provider. When I click on it nothings happens. I am following the tutorial on Youtube and was also confused to as why on his remix it says javaScript VM instead?

**Comment by Niferu** [comment]
Hello @JosephEDale 

JavaScript VM is Patrick's custom testnet I believe, but if you use Goerli it will be all good for you. Have you connected your MetaMask to Remix? Have you picked Goerli as testnet on your MetaMask instead of Mainnet?

**Reply by JosephEDale** [reply]
Hello @Neftyr 

I appreciate the help. I have not connected my MetaMask to Remix, how do I do this?

**Reply by Niferu** [reply]
Hi @JosephEDale 
Have you picked Goerli in your MetaMask account as network? It is the only working testnet now. If you are setup correctly then follow cromewar advice.
![image](https://user-images.githubusercontent.com/110537615/198622780-f5f9b711-2b03-466b-862e-3034ff799618.png)

**Comment by cromewar** [comment]
Hello @JosephEDale 
This is a known bug and it happens due to several reasons, please follow some of this recommendations:
1. Delete all the cache, cookies and navigation history of your browser.
2. Try another browser like brave or Firefox.
